lib/kino/debug.ex

```diff
@@ -0,0 +1,125 @@
+defmodule Kino.Debug do
+  @moduledoc false
+
+  @registry Kino.Debug.Registry
+
+  @doc """
+  Custom backend for `Kernel.dbg/2`.
+
+  The custom backend provides a more interactive user interface for
+  `Kernel.dbg/2` calls in certain cases, such as call pipelines. It
+  falls back to the default backend otherwise.
+  """
+  @spec dbg(Macro.t(), Macro.t(), Macro.Env.t()) :: Macro.t()
+  def dbg(ast, options, %Macro.Env{} = env) do
+    default_dbg_ast = Macro.dbg(ast, options, env)
+    dbg_id = System.unique_integer()
+
+    case ast do
+      {:|>, _meta, _args} ->
+        quote do
+          if pid = unquote(__MODULE__).lookup_dbg_handler(unquote(dbg_id)) do
```

Comment:
Quick question: what happens if `dbg` was invoked from a separate process? Do we also want to show the "fancy" version?

Reply:
Actually that's an interesting case, I would say there is no reason to differentiate that, however the relevant detail is that the started kino is tied to the process lifetime, so when debugging a process that crashes this is an issue.